controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -252,15 +252,21 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, kcp *cont
 	case numMachines < desiredReplicas && numMachines == 0:
```

Comment:
Just to double check, this allows folks to reduce the control plane replicas to 0, and when scaling up, it'd re-init the cluster. Is that intended behavior?

Reply:
It is intended behavior in that the only field we currently have to determine if the control plane has previously been initialized is `kcp.Status.Initialized` (or possibly `cluster.Status.Initialized`). If we want to ensure stronger safeguards against a second initialization, we'd need to define how we would want to do that and I'd consider it out of scope for this PR.

Reply:
Yeah definitely, was just going through the logic and noticed it, we can tackle this later!